website/blog/2025-07-01-mlflow-prompt-evaluate/index.md

```diff
@@ -0,0 +1,435 @@
+
+There are a few types of OCR, ranging from Simple to Intelligent. Today we will be focusing on the latter. This incorporates AI and ML for increased capabilities like handwriting recognition and adaptation to new styles.
+
+Fun fact - the earliest form of OCR was introduced in 1914. It was a device designed for blind indivuduals to read printed text without Braille called the Optophone.
+![Optophone](/img/blog/prompt-evaluate/margaret-hogan-used-the-black-sounding-optophone-to-read-a-book.jpg)
+
+## The Challenge with Prompting
+
+A good prompt engineer knows how to combine analytical thinking with creativity in order to get the most out of their GenAI applications. This is all while ensuring alignment with the high level goals of an organization. To refine, iterate, and automate their process, they often need the technical skills, or simply the additional effort, to do so. How can we allow prompt engineers to focus on the crafting of prompts, and less on the maintenance?
+
+There are also an abundance of challenges related verifying and assessing the expected output. Did the prompt address the complexity of the situation? Did it have the appropriate context? Too much context? Did it introduce bias?
+
+Having a robust way to iterate on prompts and assess their accuracy becomes paramount in building production-ready GenAI applications.
+
+## MLflow Key Capabilities
+
+### MLflow Prompt Registry
+
+[MLflow Prompt Registry](https://mlflow.org/docs/latest/genai/prompt-version-mgmt/prompt-registry) addresses many of the challenges prompt engineers face today. With a Git-inspired set up, prompt engineers have a streamlined way to version, track, and reuse prompts. Let's look at a typical flow.
+
+**1. Create and Register Your Prompt:** Each prompt is registered as a **Prompt Object**. This becomes a versioned, paramaterized template that can be dynamically filled at runtime. The object will include a **name** (unique identifier), **template text**, **version #**, **commit message**, **metadata/tags** (optional), and **aliases**.  
+Register your prompt with -
+`mlflow.genai.register_prompt()`
+
+**2. Manage Aliases:** Aliases ensure prompts remain aligned with their intended use. For example, "Production" and "Staging" aliases tie them to their respective environments, simplifying traceability, orchestration, and governance of our prompts.
+
+**3. Load and Use Prompts:** We are now ready to load in our prompt and start using it with our GenAI applications, like our LLM-backed OCR tool.  
+Load in your prompt with -
+`mlflow.genai.load_prompt()`
+
+**4. Iterate on Prompts:** An important step in our process is improving and updating prompts. This can be done any number of times while moving toward the accuracy required, all while staying organized in a centralized repository for prompt management.  
+<br/>
+
```

Comment:
The value prop to highlight is we can outsource "cheaper" prompt optimization to non-technical SMEs